### PR TITLE
fix: Switch to cosmjs primitives & amino offline signing logic 

### PR DESCRIPTION
**Pull request by jay-dee7**
Replace any Lum-Network SDK dependencies and use the standard cosmjs packages.
Also fixed:
- Msg validation
- package size
- msg decoding
- signature verification

### CODE REVIEW

File: src/api/constants.ts [hunk 1]
```diff
@@ -322,4 +320,5 @@ export const VC_REMOVE_ORIGINAL_FIELDS = false
 // Auth
 export const PROPOSAL_MESSAGE_TITLE = 'AuthRequest'
 
-export const REPLY_PROTECTION_INTERVAL = 30
+// since date in javascript is always millisecond
+export const REPLY_PROTECTION_INTERVAL = 30 * 1000
```
Comment (Eengineer1) [inline src/api/constants.ts:324]: @jay-dee7 Let's consider halving this to 15''.

File: src/api/controllers/authentication.ts [hunk 2]
```diff
@@ -1,170 +1,129 @@
...
-    ), 
-    {
-      headers: {
-        'content-type': 'application/json;charset=UTF-8',
-      },
-    }
-  )
+export const enum AuthTokenErrorEnum {
+	InvalidToken = "provided token is invalid",
+	NoPubkey = "public is not present inside the message body",
```
Comment (Eengineer1) [inline src/api/controllers/authentication.ts:123]: -> public **key** is not present inside the message body

File: src/api/controllers/authentication.ts [hunk 3]
```diff
@@ -1,170 +1,129 @@
...
+	TitleMismatch = "message title does not match",
+	TokenIsTooOld = "the token provided is older that 30s",
```
Comment (Eengineer1) [inline src/api/controllers/authentication.ts:125]: -> the token provided is stale

File: src/api/controllers/crypto_box.ts [hunk 4]
```diff
@@ -1,97 +1,80 @@
...
-        }
-    }
-    
-    async handlePostKVStore(request: Request): Promise<Response>{
-        return await this.putToKVStore(request)
-    }
-}
\ No newline at end of file
+	async handlePostKVStore(request: Request): Promise<Response> {
+		console.debug('this is how we roll')
```
Comment (Eengineer1) [inline src/api/controllers/crypto_box.ts:77]: - How do we roll?  🤔

File: src/api/helpers/registry.ts [hunk 5]
```diff
@@ -1,82 +1,32 @@
...
+
+function createDefaultTypes(prefix: string): AminoConverters {
+	return {
+		...createGovAminoConverters(),
+	};
 }
 
-export const LumAminoRegistry = new AminoTypes(aminoConverters);
-export const LumRegistry = new ExtendedRegistry();
+export const CheqdAminoRegistry = new AminoTypes(createDefaultTypes("cheq"));
```
Comment (Eengineer1) [inline src/api/helpers/registry.ts:31]: - Is the "cheq" prefix correct?

File: src/api/helpers/utils.ts [hunk 6]
```diff
@@ -38,33 +15,32 @@ export const generateAuthInfoBytes = (docSigners: DocSigner[], fee: Fee, signMod
...
+	});
 
-/**
- * Generate transaction sign doc bytes used to sign the transaction
- *
- * @param signDoc sign doc (as generated by the generateSignDoc function)
- */
-export const generateSignDocBytes = (signDoc: SignDoc): Uint8Array => {
-    return makeSignBytes(signDoc);
+	const signers = doc.signers.map(signer => {
```
Comment (Eengineer1) [inline src/api/helpers/utils.ts:31]: - Do we *really* need these `generateXXX` wrapper functions?

File: src/error_handler.ts [hunk 7]
```diff
@@ -1,6 +1,25 @@
...
+
+	constructor(msg: any, method?: string, statusCode?: number) {
+		super()
+		this.message = msg
+		if (method) this.method = method
+		if (statusCode) this.statusCode = statusCode
+	}
+
+	// use generic maybe? or try method overriding with declaration merging
+	static throw(opts: { msg: string, method?: string, statusCode?: number, headers?: HeadersInit }): Response {
```
Comment (Eengineer1) [inline src/error_handler.ts:16]: @jay-dee7 This method should be responsible of throwing server-side `Error` instances or make responses.

I suggest the following approach:

- Add a `makeResponse` method.
- Wrap `throw` under `throwOrRespond`.
- Add generics to `throwOrRespond`.
- Add param validation to `throwOrRespond`.
- Handle execution based on the validation outcome through either `throw` or `makeResponse`.

File: src/api/controllers/authentication.ts [hunk 8]
```diff
@@ -1,170 +1,129 @@
...
+
+	// Get sign mode
+	const signMode = decoded.authInfo.signerInfos[0].modeInfo?.single?.mode;
+	if (!signMode) {
+		return ErrorHandler.throw({ msg: AuthTokenErrorEnum.NoSignMode })
+	}
+
+	// Check the signature
+	let ok = false;
+	if (signMode === SignMode.SIGN_MODE_DIRECT) {
```
Comment (Eengineer1) [inline src/api/controllers/authentication.ts:59]: Let's utilize a `switch` statement instead, that throws by default case.

File: src/api/controllers/authentication.ts [hunk 9]
```diff
@@ -1,170 +1,129 @@
...
-    gas: _gas.toString()
-  };
-}
+export const handleAuthToken = async (token: string): Promise<Response | void> => {
+	const bz = fromBase64(token)
+	const chainId = 'cheqd-mainnet-1';
+	const decoded = decodeTxRaw(bz);
+
+	// check is msg title is what we expect and the date in msg is less than 30s old (which is 30000ms in JS/TS world)
+	const err: Response | void = checkMsg(decoded)
```
Comment (Eengineer1) [inline src/api/controllers/authentication.ts:37]: Let's not pass errors by value and switch to `ErrorHandler` logic.

**Review by Eengineer1 — CHANGES_REQUESTED** [review]
@jay-dee7 Looks good so far.
I've added a few comments on minor refactors & improvements needed.
Let me know what you think.

Thanks!

File: src/error_handler.ts (hunk 7 above)
Comment (jay-dee7) [reply src/error_handler.ts:16]: @Eengineer1 can we do this in another PR? I'd also like to tackle the logging stuff

File: src/api/constants.ts (hunk 1 above)
Comment (jay-dee7) [reply src/api/constants.ts]: done

File: src/api/controllers/authentication.ts (hunk 3 above)
Comment (jay-dee7) [reply src/api/controllers/authentication.ts]: done

File: src/error_handler.ts (hunk 7 above)
Comment (Eengineer1) [reply src/error_handler.ts:16]: @jay-dee7 Sounds good.

**Review by Eengineer1 — APPROVED** [review]
@jay-dee7 Approved with minor corrections.
Will let you resolve `package.json` & lock file dependency conflicts before merging.

Thanks!

File: package.json [hunk 10]
```diff
@@ -27,52 +27,45 @@
...
     "eslint-config-typescript": "^3.0.0",
-    "jest": "^27.5.1",
-    "miniflare": "^2.7.1",
-    "prettier": "^2.7.1",
-    "semantic-release": "^19.0.5",
+    "jest": "^29.0.2",
+    "miniflare": "^2.4.0",
+    "prettier": "^2.3.0",
     "service-worker-mock": "^2.0.5",
-    "terser-webpack-plugin": "^5.3.5",
```
Comment (Eengineer1) [inline package.json:50]: @jay-dee7 Why are we accepting older versions?

File: package.json [hunk 11]
```diff
@@ -27,52 +27,45 @@
...
     "path-browserify": "1.0.1",
-    "process": "^0.11.10",
+    "buffer": "6.0.3",
+	"eslint-plugin-jest": "^27.0.1",
+	"eslint-plugin-prettier": "^4.2.1",
     "stream-browserify": "3.0.0",
+    "process": "^0.11.10",
     "util": "^0.12.4"
   },
-  "publishConfig": {
```
Comment (Eengineer1) [inline package.json:74]: @jay-dee7 We probably need this to configure the registry scope as well on release.

**Review by Eengineer1 — CHANGES_REQUESTED** [review]
@jay-dee7 Requested slight mods and this is good to go.
Thanks!

**Comment by Eengineer1** [comment]
@jay-dee7 Don't bother with the PR formatting, as linting rule references should be updated by merging this PR.

File: package.json (hunk 10 above)
Comment (jay-dee7) [reply package.json:50]: this is accidental, I missed it when doing a rebase. I'll fix it right away

File: package.json (hunk 11 above)
Comment (jay-dee7) [reply package.json:74]: totally accidental

File: package.json [hunk 12]
```diff
@@ -27,49 +27,54 @@
...
     "terser-webpack-plugin": "^5.3.5",
-    "ts-jest": "^27.1.5",
-    "ts-loader": "^9.3.1",
-    "typescript": "^4.7.4",
-    "webpack": "^5.74.0",
-    "webpack-cli": "^4.10.0"
+    "ts-jest": "29.0.0-next.1",
+    "ts-loader": "^9.2.2",
+    "typescript": "^4.3.2",
+    "webpack": "^5.38.1",
```
Comment (Eengineer1) [inline package.json:54]: @jay-dee7 This bundle as well.

File: package.json [hunk 13]
```diff
@@ -27,49 +27,54 @@
...
+    "webpack-bundle-analyzer": "^4.6.1",
+    "webpack-cli": "^4.7.0",
```
Comment (Eengineer1) [inline package.json:56]: @jay-dee7 And this.

**Review by Eengineer1 — APPROVED** [review]
Approved.

**Pull request merged**